src/lib/expression/between_expression.hpp

```diff
@@ -6,15 +6,26 @@
 
 namespace opossum {
 
-class BetweenExpression : public AbstractPredicateExpression {
+class BaseBetweenExpression : public AbstractPredicateExpression {
```

Comment:
As far as I can tell there is no need to have four(!) distinct classes for Between expressions. We cluster similar expressions into the same expression-class - and just as >, >=, <, ... are all "BinaryPredicateExpression", all four between-types are "BetweenExpressions". Please revert this back to a single BetweenExpression-class.